src/Fractal/ResourceTransformer.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Giadc\JsonApiResponse\Fractal;
+
+use Giadc\JsonApiRequest\Requests\RequestParams;
+use Giadc\JsonApiResponse\Interfaces\AbstractDataModifier;
+use Giadc\JsonApiResponse\Interfaces\JsonApiResource;
+use League\Fractal\TransformerAbstract;
+use Symfony\Component\HttpFoundation\Request;
+
+abstract class ResourceTransformer extends TransformerAbstract
+{
+    protected RequestParams $requestParams;
+
+    protected array $defaultFields = [];
```

Comment:
Not fully-formed impression:

This can get very repetitive, with `$defaultFields` just being a duplication of most of the keys returned from a `jsonSerialize()` function. Do you think it could possibly useful if a `ResourceTransformer` had its own `fields` and `excludes` functionality? It may be less verbose to simply list the fields you _don't_ want included.

Counterpoint to my own point: being verbose helps avoid unexpected data leaks.

Reply:
I've been split on this myself...So I just defaulted to being more verbose for that reason.

Reply:
I think I'm cool with the verbosity. It's still better than our previous `transform()` functions that needed to manually pick out data.